openrtx/include/core/user_functions.h

```diff
@@ -0,0 +1,60 @@
+/***************************************************************************
+ *   Copyright (C) 2020 - 2024 by Federico Amedeo Izzo IU2NUO,             *
+ *                                Niccolò Izzo IU2KIN,                     *
+ *                                Silvano Seva IU2KWO                      *
+ *                                Morgan Diepart ON4MOD                    *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 3 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>   *
+ ***************************************************************************/
+
+#ifndef USER_FUNCTIONS_H
+#define USER_FUNCTIONS_H
+
+#include <hwconfig.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <errno.h>
+
+#if defined(CONFIG_USER_FUNCTIONS) && (CONFIG_USER_FUNCTIONS > 0)
```

Comment:
I think this guard is not needed, because:
- target-specific code using the user function system, will for sure need to define the `CONFIG_USER_FUNCTIONS`
- all the function calls in the common code have to be guarded with `#ifdef` to avoid linker errors in case the target does not support the user functions (see threads.c)
- functions not called anywhere and unused variables are garbage-collected at link time

Reply:
I can remove the guard if needed. But if user functions are expected to be used in common code then it is not really user functions anymore and we should also modify slightly the mechanism to account for functions added from common code.

Reply:
I called them "user functions" following the terminology used in the code I used at work, but a more correct term should be "tasklet" or something like that. About adding periodic functions from common code, I see that the current `gps_task` is a good candidate for this mechanism: if there is the GPS hardware, or if the GPS is enabled, add a periodic function which retrieves and parses the NMEA sentences.